File: src/types/media.ts
/**
 * Media Types - Video Links and Trello Cards
 * Feature: 004-embed-multiple-video
 */

/**
 * Represents a video link (typically Sprout Video) associated with a project
 */
export interface VideoLink {
  /** Full video URL (e.g., https://sproutvideo.com/videos/abc123) */
  url: string

  /** Extracted Sprout Video ID (e.g., "abc123") */
  sproutVideoId?: string

  /** User-provided or fetched video title */
  title: string

  /** Cached thumbnail URL from Sprout API */
  thumbnailUrl?: string

  /** ISO 8601 timestamp of upload */
  uploadDate?: string

  /** Original filename from Renders/ folder */
  sourceRenderFile?: string
}

/**
 * Represents a Trello card associated with a project
 */
export interface TrelloCard {
  /** Full Trello card URL (e.g., https://trello.com/c/abc123/project-name) */
  url: string

  /** Extracted card ID (e.g., "abc123") */
  cardId: string

  /** Fetched card name/title from Trello API */
  title: string

  /** Optional board name from Trello API */
  boardName?: string

  /** ISO 8601 timestamp of last title fetch */
  lastFetched?: string
}

/**
 * Represents a Trello board with metadata
 * Returned from fetch_trello_boards Tauri command
 */
export interface TrelloBoard {
  /** Board ID (24-character alphanumeric) */
  id: string

  /** Board display name */
  name: string

  /** Organization details (if board belongs to an org) */
  organization?: TrelloOrganization

  /** Board preferences including visibility */
  prefs: TrelloBoardPrefs
}

/**
 * Trello organization information
 */
export interface TrelloOrganization {
  /** Organization display name */
  name: string
}

/**
 * Trello board preferences
 */
export interface TrelloBoardPrefs {
  /** Board permission level (e.g., "public", "private", "org") */
  permissionLevel: string
}

/**
 * Sprout Video API response structure
 * Returned from fetch_sprout_video_details Tauri command
 */
export interface SproutVideoDetails {
  /** Sprout Video ID */
  id: string

  /** Video title from Sprout API */
  title: string

  /** Video description (optional) */
  description?: string

  /** Video duration in seconds (floating point for precision) */
  duration: number

  /** Asset URLs including thumbnails */
  assets: SproutAssets

  /** ISO 8601 timestamp of video creation */
  created_at: string
}

/**
 * Sprout Video assets structure
 */
export interface SproutAssets {
  /** Array of poster frame/thumbnail URLs */
  poster_frames: string[]
}

File: src-tauri/src/media.rs
use serde::{Deserialize, Serialize};

/// Represents a video link (typically Sprout Video) associated with a project
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct VideoLink {
    /// Full video URL (e.g., https://sproutvideo.com/videos/abc123)
    pub url: String,

    /// Extracted Sprout Video ID (e.g., "abc123")
    #[serde(rename = "sproutVideoId", skip_serializing_if = "Option::is_none")]
    pub sprout_video_id: Option<String>,

    /// User-provided or fetched video title
    pub title: String,

    /// Cached thumbnail URL from Sprout API
    #[serde(rename = "thumbnailUrl", skip_serializing_if = "Option::is_none")]
    pub thumbnail_url: Option<String>,

    /// ISO 8601 timestamp of upload
    #[serde(rename = "uploadDate", skip_serializing_if = "Option::is_none")]
    pub upload_date: Option<String>,

    /// Original filename from Renders/ folder
    #[serde(rename = "sourceRenderFile", skip_serializing_if = "Option::is_none")]
    pub source_render_file: Option<String>,
}

/// Represents a Trello card associated with a project
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TrelloCard {
    /// Full Trello card URL (e.g., https://trello.com/c/abc123/project-name)
    pub url: String,

    /// Extracted card ID (e.g., "abc123")
    #[serde(rename = "cardId")]
    pub card_id: String,

    /// Fetched card name/title from Trello API
    pub title: String,

    /// Optional board name from Trello API
    #[serde(rename = "boardName", skip_serializing_if = "Option::is_none")]
    pub board_name: Option<String>,

    /// ISO 8601 timestamp of last title fetch
    #[serde(rename = "lastFetched", skip_serializing_if = "Option::is_none")]
    pub last_fetched: Option<String>,
}

/// Sprout Video API response structure
/// Returned from fetch_sprout_video_details command
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SproutVideoDetails {
    /// Sprout Video ID
    pub id: String,

    /// Video title from Sprout API
    pub title: String,

    /// Video description (optional)
    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,

    /// Video duration in seconds (floating point for precision)
    pub duration: f64,

    /// Asset URLs including thumbnails
    pub assets: SproutAssets,

    /// ISO 8601 timestamp of video creation
    pub created_at: String,
}

/// Sprout Video assets structure
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SproutAssets {
    /// Array of poster frame/thumbnail URLs
    pub poster_frames: Vec<String>,
}